controllers/etcd_controller.go

```diff
@@ -515,23 +519,29 @@ func (r *EtcdReconciler) syncConfigMapData(cm *corev1.ConfigMap, etcd *druidv1al
 	return cmCopy, err
 }
 
-func (r *EtcdReconciler) getConfigMapFromEtcd(etcd *druidv1alpha1.Etcd) (*corev1.ConfigMap, error) {
+func (r *EtcdReconciler) getConfigMapFromEtcd(etcd *druidv1alpha1.Etcd, values *map[string]interface{}) (*corev1.ConfigMap, error) {
 	var err error
 	decoded := &corev1.ConfigMap{}
 	configMapPath := getChartPathForConfigMap()
-	if _, ok := r.RenderedChart.Files()[configMapPath]; !ok {
+	chartPath := getChartPath()
+	renderedChart, err := r.ChartApplier.Render(chartPath, etcd.Name, etcd.Namespace, *values)
```

Comment:
Why render again and again? Why not do it once and pass it as a parameter to the func?

Reply:
We have to refactor extensively again. I shall optimise this in that cycle.